render/render.js

```diff
@@ -489,18 +489,18 @@ module.exports = function($window) {
 	}
 
 	//lifecycle
-	function initLifecycle(source, vnode, hooks) {
-		if (typeof source.oninit === "function") source.oninit.call(vnode.state, vnode)
-		if (typeof source.oncreate === "function") hooks.push(source.oncreate.bind(vnode.state, vnode))
+	function initLifecycle(vnode, source, hooks) {
```

Comment:
At last, if class and closure components are baked into `createComponent()`, the need for setting `state` in `oninit` vanishes and the old `*Lifecycle()` code can be restored.